src/chemcaption/featurize/utils.py

```
@@ -23,13 +23,25 @@
-    """Join list elements into a string. First elements separated by comma, last element separated by `and`."""
-    if len(elements) == 1:
-        return str(elements[0])
-
-    return ", ".join([str(e) for e in elements[:-1]]) + ", and " + str(elements[-1])
 
+def join_list_elements(names: List[str], elements: Any) -> str:
+    """Join list elements into a string. All elements separated by comma."""
+    
+    parts = []
+
+    for name, amount in zip(names, elements):
```

Ruff (0.15.12)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)